### PR TITLE
Final

### PR DESCRIPTION
**Pull request by marcelomaidden**
## Final
At this pull request, I add functionality for the buttons:
- Add clickHandler to perform math operations (+, -, X, /, %, +/-)
- Add style to the buttons passing the className via props
- Check if the last clicked button is an operation
- Use props and state to update the Virtual DOM
- Use big.js to handle big numbers

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -1,15 +1,67 @@
...
+      lastClicked: null,
+    };
+  }
+
+  handleClick = buttonName => {
+    const {
+      total, next, operation, lastClicked,
+    } = this.state;
+    const operations = ['+', '-', '÷', 'X', '='];
+    console.log(lastClicked);
```
Comment (Forison) [inline src/components/App.js]: let remove `console.log`, this is not a good practice as you may end up sending sensitive info.

File: src/components/App.js [hunk 2]
```diff
@@ -1,15 +1,67 @@
...
+    if (buttonName === '.') {
+      this.setState({
```
Comment (Forison) [inline src/components/App.js]: let move this code calculate.js, handleclick would then rely on the logic in the calculate.js.

**Review by Forison — CHANGES_REQUESTED** [review]
## Status:: Change Required ❌

# Hi Marcelo

#### You have done a good job on this one so far, Good job on styling the application even though it is not a requirement🤗.

* [x] The image in the in Readme is not showing.

* [x] Kindly remove all `console.log` from the code.

* [ ] Kindly move the logic in the handleClick to calculate.js

* [x] When a number is divided by zero, it should return `undefined`, not zero.

* [x] When the modifier `+/-` is selected and number1 and number two are present, it should mutate both numbers not one. ie `4 + 5` after selecting `+/-` should become `-4 - 5` not `4-5`

**[OPTIONAL]**
The user experience could be improved by allowing users see both operand and the operator whilst performing a calculation.

#### That is it, kindly make the changes and submit a re-review😎.

#### Happy coding👨‍💻

Addo Forison

**Comment by marcelomaidden** [comment]
Hello Forison. Thanks for pointing out where I need to improve my code. I would like to mention that '+/-' is already converting both numbers to the reverse (positive, negative). Together with this message, I send a screenshot from the code for conference.
Screenshot from 2021-02-11 18-50-58.

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Changes Required

### Hi @marcelomaidden :wave: 

The **+/-**  operation is not working properly; It does not work for division and multiplication. I will leave a git for reference. I also discovered that operation like(-8-2) returns +6 instead of -10.

gif1


Make the change then submit another form for a review.

### Have a great day :slightly_smiling_face: 

#### ``` Uchechukwu Henry Inyama ```

File: src/components/Display.js [hunk 3]
```diff
@@ -1,14 +1,30 @@
...
+  );
 };
 
 Display.propTypes = {
-  result: PropTypes.string,
+  result: PropTypes.shape({
+    total: PropTypes.string,
+    next: PropTypes.string,
+    operation: PropTypes.string,
+  }).isRequired,
```
Comment (christianotieno) [inline src/components/Display.js]: ## Fix this prop-error validation
- [x] Hello, we have some informing errors in the console. Kindly fix these prop validations.
prop-error

**Review by christianotieno — CHANGES_REQUESTED** [review]
## Changes Required ♻️ 

Hello @marcelomaidden 👋 
We have some few issues in the console, React is complaining about in which I have highlighted down below 👇 
### Happy coding 💻 
#### Christian Otieno (CR)

File: src/components/Display.js [hunk 4]
```diff
@@ -21,9 +21,9 @@ const Display = ({ result }) => {
...
 Display.propTypes = {
   result: PropTypes.shape({
-    total: PropTypes.string,
-    next: PropTypes.string,
-    operation: PropTypes.string,
+    total: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    next: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    operation: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
   }).isRequired,
 };
```
Comment (shloch) [inline src/components/Display.js]: Awesome job, straight to the point

**Review by shloch — APPROVED** [review]
## Hello @marcelomaidden 

This issue is diligently solved and based on that you get thumbs for the next step
accomplished

HAPPY CODING

File: src/components/Display.js (hunk 4 above)
Comment (marcelomaidden) [reply src/components/Display.js]: Thanks to all code reviewers.

**Pull request merged**